spark-knn-core/src/main/scala/org/apache/spark/ml/knn/KNN.scala

```diff
@@ -228,12 +265,16 @@ class KNNModel private[ml](
   def setBufferSize(value: Double): this.type = set(bufferSize, value)
 
   //TODO: All these can benefit from DataSet API in Spark 1.6
-  override def transform(dataset: DataFrame): DataFrame = {
-    val merged = transform(dataset, topTree, subTrees)
+  override def transform(dataset: Dataset[_]): DataFrame = transform(dataset, withDistance = false)
```

Comment:
I don't really like the `DummyImplicit` hack and have two function with the same name do two different things. Can `withDistance` be a parameter instead? I recall there are already other params that only apply to prediction already?